Building HTML fails (also on docker)

My attempts at moving my [bookbook book](https://fabienmaussion.info/scientific_programming) to jupyter-book are advancing slowly ;-).

I have the following build issue with docker (from your [documentation](https://jupyter.org/jupyter-book/guide/04_publish.html), note that the link to the troubleshooting FAQ is broken):

```
$ docker run --rm --security-opt label:disable  \
>    -v /home/mowglie/tmp/jbook/mybookname:/srv/jekyll \
>    -p 4000:4000 \
>    -it -u 1000:1000 \
>    emdupre/jupyter-book bundle exec jekyll serve --host 0.0.0.0
Could not find concurrent-ruby-1.1.5 in any of the sources
```

Using the conda-forge package (`rb-github-pages`), I get:

```
$ make serve
bundle exec guard
Configuration file: _config.yml
CiteProc requires the `unicode_utils` or `unicode` Gem on Ruby 2.3
19:01:53 - ERROR - Invalid Guardfile, original error is: 
> [#] 
> [#] undefined method `upcase' for module `CiteProc', 
> [#] backtrace: 
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-1.0.9/lib/citeproc/compatibility.rb:63:in `module_function'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-1.0.9/lib/citeproc/compatibility.rb:63:in `<module:CiteProc>'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-1.0.9/lib/citeproc/compatibility.rb:62:in `<top (required)>'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-1.0.9/lib/citeproc.rb:32:in `require'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-1.0.9/lib/citeproc.rb:32:in `<top (required)>'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-ruby-1.1.10/lib/citeproc/ruby.rb:4:in `require'
> [#] 	(dsl)> $GEM_HOME/gems/citeproc-ruby-1.1.10/lib/citeproc/ruby.rb:4:in `<top (required)>'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-scholar-5.16.0/lib/jekyll/scholar.rb:7:in `require'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-scholar-5.16.0/lib/jekyll/scholar.rb:7:in `<top (required)>'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-scholar-5.16.0/lib/jekyll-scholar.rb:1:in `require'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-scholar-5.16.0/lib/jekyll-scholar.rb:1:in `<top (required)>'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/external.rb:58:in `require'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/external.rb:58:in `block in require_with_graceful_fail'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/external.rb:55:in `each'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/external.rb:55:in `require_with_graceful_fail'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/plugin_manager.rb:30:in `require_gems'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/plugin_manager.rb:22:in `conscientious_require'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/site.rb:114:in `setup'
> [#] 	(dsl)> $GEM_HOME/gems/jekyll-3.8.5/lib/jekyll/site.rb:32:in `initialize'
> [#] 	(dsl)> $GEM_HOME/gems/guard-jekyll-plus-2.0.2/lib/guard/jekyll_plus/builder.rb:20:in `new'
> [#] 	(dsl)> $GEM_HOME/gems/guard-jekyll-plus-2.0.2/lib/guard/jekyll_plus/builder.rb:20:in `reload'
> [#] 	(dsl)> $GEM_HOME/gems/guard-jekyll-plus-2.0.2/lib/guard/jekyll_plus/builder.rb:15:in `initialize'
> [#] 	(dsl)> $GEM_HOME/gems/guard-jekyll-plus-2.0.2/lib/guard/jekyll_plus.rb:20:in `new'
> [#] 	(dsl)> $GEM_HOME/gems/guard-jekyll-plus-2.0.2/lib/guard/jekyll_plus.rb:20:in `initialize'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/plugin_util.rb:62:in `new'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/plugin_util.rb:62:in `initialize_plugin'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/internals/plugins.rb:26:in `add'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/dsl.rb:185:in `block in guard'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/dsl.rb:182:in `each'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/dsl.rb:182:in `guard'
> [#] 	(dsl)> ./Guardfile:1:in `evaluate'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/dsl.rb:377:in `instance_eval'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/dsl.rb:377:in `evaluate'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/guardfile/evaluator.rb:89:in `evaluate'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard.rb:134:in `_evaluate'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard.rb:49:in `setup'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/commander.rb:32:in `start'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/cli/environments/valid.rb:16:in `start_guard'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/cli.rb:122:in `start'
> [#] 	(dsl)> $GEM_HOME/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
> [#] 	(dsl)> $GEM_HOME/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
> [#] 	(dsl)> $GEM_HOME/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
> [#] 	(dsl)> $GEM_HOME/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/aruba_adapter.rb:32:in `execute'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/lib/guard/aruba_adapter.rb:19:in `execute!'
> [#] 	(dsl)> $GEM_HOME/gems/guard-2.14.2/bin/_guard-core:11:in `<main>'
Makefile:30: recipe for target 'serve' failed
make: *** [serve] Error 1
```

Any idea?

I think I understand better why @choldgraf tries to move away from jekyll :roll_eyes:

--- a/jupyter_book/Dockerfile
+++ b/jupyter_book/Dockerfile
@@ -2,4 +2,4 @@ FROM jekyll/jekyll:3.8.5
 COPY book_template/Gemfile .
 RUN apk --update add make gcc g++ libc-dev
 RUN gem install bundler
-RUN bundle install
+ENTRYPOINT bundle install && bundle exec jekyll serve --host 0.0.0.0

--- a/jupyter_book/book_template/content/guide/04_publish.md
+++ b/jupyter_book/book_template/content/guide/04_publish.md
@@ -214,7 +214,7 @@ docker run --rm --security-opt label:disable  \
    -v /full/path/to/your/book:/srv/jekyll \
    -p 4000:4000 \
    -it -u 1000:1000 \
-   emdupre/jupyter-book bundle exec jekyll serve --host 0.0.0.0
+   emdupre/jupyter-book
 ```
 
 If you navigate to `http://0.0.0.0:4000/jupyter-book/` in your browser,
@@ -245,7 +245,7 @@ Make sure to specify the full path to your Jupyter Book, rather than the relativ
 ```bash
 singularity run -B /full/path/to/your/book:/srv/jekyll \
     --pwd /srv/jekyll \
-    jupyter-book.simg bundle exec jekyll serve
+    jupyter-book.simg
 ```
 
 And that's it! If you navigate to `http://127.0.0.1:4000/jupyter-book/` in your browser,

